### PR TITLE
Backported getLocalTime from ESP32 Arduino

### PR DESCRIPTION
**Pull request by h1aji**
New PR. Removed previosly added modifications.
This time just importing new getLocalTime from ESP32 Arduino core

### CODE REVIEW

**Comment by d-a-v** [comment]
It seems that you have to rename or remove [this one](https://github.com/esp8266/Arduino/blob/55ef3e73978ad010d44ad7e4c9bd4b4f32455989/libraries/LittleFS/examples/LittleFS_Timestamp/LittleFS_Timestamp.ino#L22-L42)

**Comment by h1aji** [comment]
the only part that appears different is -  ` uint32_t count = ms / 10;`

I assume we can modify https://github.com/esp8266/Arduino/blob/55ef3e73978ad010d44ad7e4c9bd4b4f32455989/libraries/LittleFS/examples/LittleFS_Timestamp/LittleFS_Timestamp.ino#L156 and make it 
`  getLocalTime(&tmstruct, 500);
`
then delete **getLocalTime** from that sketch

@earlephilhower what do you think?

**Comment by h1aji** [comment]
@d-a-v does this look ok now?

File: cores/esp8266/time.cpp [hunk 1]
```diff
@@ -198,15 +198,43 @@ void setTZ(const char* tz){
...
+ 
+  time(&now);
+  localtime_r(&now, info);
+ 
+  if (info->tm_year > (2016 - 1900)) 
+  {
+    return true;
+  }
+ 
+  while (count--)
```
Comment (d-a-v) [inline cores/esp8266/time.cpp:225]: With this loop, maximum timeout is 10 times the value of `ms`, which is why Earle divided it by 10.
Original and probably later updated / fixed esp32 code [is there](https://github.com/espressif/arduino-esp32/blob/34125cee1d1c6a81cd1da07de27ce69d851f9389/cores/esp32/esp32-hal-time.c#L91) with a more accurate loop.
Do you think you can improve your proposal ?

**Comment by d-a-v** [comment]
About the CI error, `time.cpp` is not included in host tests because it holds standard functions already available on host's system.
I suggest moving the function into a new file (`timehelper.cpp`?) and adding this file in the [host build Makefile](https://github.com/esp8266/Arduino/blob/42aa0e6d253603a136cd813b7b834364791b89ad/tests/host/Makefile#L99-L100).

File: cores/esp8266/time.cpp (hunk 1 above)
Comment (h1aji) [reply cores/esp8266/time.cpp]: Okay then, do you have an idea how LittleFS_Timestamp.ino sketch should be modified?

**Comment by h1aji** [comment]
It looks like this PR causing lots of troubles. Shall I close it?

**Comment by d-a-v** [comment]
> Shall I close it?

No :) if you want to do it.

File: cores/esp8266/time.cpp (hunk 1 above)
Comment (d-a-v) [reply cores/esp8266/time.cpp]: I think it is OK to use the esp32 implementation globally and remove littlefs test's one.

**Comment by h1aji** [comment]
I dont mind if you add it yourself, otherwise I am happy with this commit, I wasnt sure how to change that sketch file, that why i though asking you

**Comment by d-a-v** [comment]
> otherwise I am happy with this commit

I used updated esp32 code in #8413, moved the function into another file so it can be used for the esp8266, emulation on host, and host tests, and removed the old local copy from the littleFS test.

You can get inspiration from it to fix the CI builds if you like.

**Comment by h1aji** [comment]
@d-a-v nice. tthank you.
I will close this PR

**Pull request closed without merging**